### PR TITLE
Add support for group process monitoring

### PR DESCRIPTION
**Pull request by Pratyush8083**
**Description:** <Describe what has changed.>
Adding support for group process monitoring

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: deploy.yaml [hunk 1]
```diff
@@ -101,6 +101,12 @@ spec:
       labels:
         app: otelcontribcol
     spec:
+      volumes:
```
Comment (mithunbelur) [inline deploy.yaml:104]: I hope this volume is not required. Remove it.

File: receiver/hostmetricsreceiver/README.md [hunk 2]
```diff
@@ -38,18 +38,19 @@ hostmetrics:
...
+| [disk]         | All except Mac<sup>[1]</sup> | Disk I/O metrics                                       |
+| [load]         | All                          | CPU load metrics                                       |
+| [filesystem]   | All                          | File System utilization metrics                        |
+| [memory]       | All                          | Memory utilization metrics                             |
+| [network]      | All                          | Network interface I/O metrics & TCP connection metrics |
+| [paging]       | All                          | Paging/Swap space utilization and I/O metrics          |
+| [processes]    | Linux, Mac                   | Process count metrics                                  |
+| [process]      | Linux, Windows, Mac          | Per process CPU, Memory, and Disk I/O metrics          |
+| [system]       | Linux, Windows, Mac          | Miscellaneous system metrics                           |
+| [groupprocess] | Linux, Windows, Mac          | Per process group CPU, Memory, and Disk I/O metrics    |
```
Comment (mithunbelur) [inline receiver/hostmetricsreceiver/README.md:53]: Add a keywork aggregated metrics of all processes of group

File: receiver/hostmetricsreceiver/README.md [hunk 3]
```diff
@@ -134,6 +136,34 @@ The following settings are optional:
...
+              match_type: "strict"
+            cmdline:
+              names:
+                - "--config=/var/lib/kube-proxy/config.conf"
+              match_type: "regexp"
+```
+In this example, the groupprocessscraper is configured to scrape metrics for processes that match the specified comm, exe, and cmdline criteria. The group_name is used to identify the group of processes being monitored.
+For `comm` and `exe`, the list of strings is an `OR`, meaning any process matching any of the strings will be added to the process group.
+For `cmdline`, the list of regexes is an `AND`, meaning they all must match.
+
```
Comment (mithunbelur) [inline receiver/hostmetricsreceiver/README.md:166]: Also, add a line.
If more than one out of(comm, exe and cmdline) is given all of them has to match.

if a process matches to one group, the same process will not be part of other groups.

File: receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/process_scraper.go [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+	metricsLen = cpuMetricsLen + memoryMetricsLen + diskMetricsLen + memoryUtilizationMetricsLen + pagingMetricsLen + threadMetricsLen + contextSwitchMetricsLen + fileDescriptorMetricsLen + signalMetricsLen
+)
+
+// scraper for Process Metrics
+type scraper struct {
+	settings           receiver.Settings
+	config             *Config
+	mb                 *metadata.MetricsBuilder
+	scrapeProcessDelay time.Duration
+	ucals              map[int32]*ucal.CPUUtilizationCalculator
```
Comment (mithunbelur) [inline receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/process_scraper.go:48]: Do we need this?

File: receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/ucal/cpu_utilization_calculator.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ucal // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/ucal"
```
Comment (mithunbelur) [inline receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/ucal/cpu_utilization_calculator.go:4]: Remove this if its not used.

Comment (Pratyush8083) [reply receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/ucal/cpu_utilization_calculator.go:4]: okay, will test once after removing

Comment (Pratyush8083) [reply receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/ucal/cpu_utilization_calculator.go:4]: As per my observation, the `ucals` package is used to store instances of CPUUtilizationCalculator for different processes, which are responsible for calculating CPU utilization metrics. If we remove `ucals`, the CPU utilization metrics will not be calculated and recorded, leading to missing metrics. So let's keep this package.

Comment (Pratyush8083) [reply receiver/hostmetricsreceiver/internal/scraper/groupprocessscraper/ucal/cpu_utilization_calculator.go:4]: I have tested after removing `ucal` package and everything seems to be working fine. Hence removing this package.

**Pull request merged**